### PR TITLE
chore(proxy): extract avatar into a lib

### PR DESCRIPTION
**Pull request by rudolfs**
As requested by @cloudhead, extract `avatar.rs` into a lib so it can be shared between `radicle-upstream` and `radicle-bins`.

### CODE REVIEW

**Review by FintanH — DISMISSED** [review]
Slick :smile:

File: proxy/api/src/http/avatar.rs [hunk 1]
```diff
@@ -15,9 +15,10 @@ pub fn get_filter() -> BoxedFilter<(impl Reply,)> {
 
 /// Avatar handlers for conversion between core domain and http request fullfilment.
 mod handler {
+    use radicle_avatar as avatar;
     use warp::{http::StatusCode, reply, Rejection, Reply};
```
Comment (xla) [inline proxy/api/src/http/avatar.rs:19]:
```suggestion
    use warp::{http::StatusCode, reply, Rejection, Reply};
    
    use radicle_avatar as avatar;
```

File: proxy/api/src/http/identity.rs [hunk 2]
```diff
@@ -97,10 +97,11 @@ pub struct CreateInput {
 #[cfg(test)]
 mod test {
     use pretty_assertions::assert_eq;
+    use radicle_avatar as avatar;
     use serde_json::{json, Value};
     use warp::{http::StatusCode, test::request};
```
Comment (xla) [inline proxy/api/src/http/identity.rs:102]:
```suggestion
    use serde_json::{json, Value};
    use warp::{http::StatusCode, test::request};
    
    use radicle_avatar as avatar;
```

File: proxy/api/src/identity.rs [hunk 3]
```diff
@@ -2,7 +2,8 @@
 
 use serde::{Deserialize, Serialize};
 
-use crate::{avatar, error};
+use crate::error;
+use radicle_avatar as avatar;
```
Comment (xla) [inline proxy/api/src/identity.rs:6]:
```suggestion
use radicle_avatar as avatar;

use crate::error;
```

**Review by xla — COMMENTED** [review]
What are we going to do about the UI part? Seems it should go into the same repo and offer an npm package which can be used in conjunction.

**Comment by cloudhead** [comment]
> What are we going to do about the UI part? Seems it should go into the same repo and offer an npm package which can be used in conjunction.

I think I would have this in a `radicle-svelte-components` repo instead, because there's a lot of ways to display these avatars, and it would be nice if this was just the core logic..

**Review by cloudhead — DISMISSED** [review]

**Review by FintanH — DISMISSED** [review]

**Review by FintanH — APPROVED** [review]

**Review by cloudhead — APPROVED** [review]

**Pull request merged**